app/uk/gov/hmrc/pillar2submissionapi/controllers/UktrSubmissionController.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pillar2submissionapi.controllers
+
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.pillar2submissionapi.models.uktrsubmissions.UktrSubmission
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class UktrSubmissionController @Inject() (cc: ControllerComponents) extends BackendController(cc) {
+
+  def submitUktr(): Action[AnyContent] = Action { request =>
+    request.body.asJson match {
+      case Some(request) =>
+        if (request.validate[UktrSubmission].isError) {
+          BadRequest("Bad request!")
+        } else Ok("")
+      case None => BadRequest("No request body!")
```

Comment:
Following up on our chat about JSON validation feedback - I've implemented a basic version that at least gives developers some insight when their JSON is malformed:
```
def submitUktr(): Action[AnyContent] = Action { request =>
    request.body.asJson.fold[Result](
      BadRequest(Json.obj("message" -> "Missing or invalid JSON body"))
    ) { json =>
      json.validate[UktrSubmission] match {
        case JsSuccess(_, _) =>
          Ok(Json.obj("message" -> "Submission received successfully"))
        case JsError(errors) =>
          BadRequest(
            Json.obj(
              "message" -> "Invalid submission format",
              "errors"  -> JsError.toJson(errors)
            )
          )
      }
    }
  }
```
It's not the prettiest output:
```
{
"message": "Invalid submission format",
"errors": {
"obj.qualifyingGroup": [{"msg":["error.path.missing"],"args":[]}],
"obj.obligationDTT": [{"msg":["error.path.missing"],"args":[]}],
"obj.liabilities.liableEntities[0].idValue": [{"msg":["error.path.missing"],"args":[]}]
}
}
```
But since our users are developers, this technical feedback might be better than nothing! All of that said, this improvement might be better placed in a separate ticket. 🤔

Let me know what you think!

Reply:
It's certainly an improvement! I think some nicer wording would improve it a little.

But yes I think this type of validation is better suited for a different ticket - this is just for the 'spec' (ie. simply accept the request, return 200)
![image](https://github.com/user-attachments/assets/1f23f357-5a73-4de6-9d72-08e57d4fd8ff)


But certainly worth keeping this changes around for when the validation ticket is picked up!

Reply:
From yesterdays discussion I think we've agreed that the error handling is going to be in it's own ticket & done in it's own class/classes (see this ticket: https://jira.tools.tax.service.gov.uk/browse/PIL-1347)